src/main/java/com/saralein/server/connection/ConnectionHandler.java

```diff
@@ -1,29 +1,31 @@
 package com.saralein.server.connection;
 
 import com.saralein.server.logger.Logger;
-import com.saralein.server.response.Response;
-
+import com.saralein.server.request.Request;
+import com.saralein.server.request.RequestParser;
+import com.saralein.server.router.Router;
 import java.io.IOException;
+import java.util.HashMap;
 
 public class ConnectionHandler implements Runnable {
     private final Logger logger;
     private final Connection socket;
-    private final Response responseBuilder;
+    private final Router router;
+    private final RequestParser requestParser;
 
-    public ConnectionHandler(Connection socket, Logger logger, Response responseBuilder) {
+    public ConnectionHandler(Connection socket, Logger logger, Router router, RequestParser requestParser) {
         this.socket = socket;
         this.logger = logger;
-        this.responseBuilder = responseBuilder;
-    }
-
-    public void sendResponse(Connection socket) throws IOException {
-        byte[] response = responseBuilder.createResponse();
-        socket.write(response);
+        this.router = router;
+        this.requestParser = requestParser;
     }
 
     public void run() {
         try {
-            sendResponse(socket);
+            HashMap<String, String> parsedRequest = requestParser.parse(socket.read());
```

Comment:
`requestParser.parse` leads me to believe I'll be getting a "Request" back, since the Request is just a simple data structure with a nice interface, I think it's ok to initialize it in the `RequestParser.parse` method

Reply:
Sounds good.  [I modified it](https://github.com/saralein/java-server/pull/3/commits/ba8f296a9bf4d7fefa5a82066e225018274894d0#diff-dd61a0538ad5c97692c4f256214c22b1R12).  I had something like this originally.  Should have left it but I like it more with it back.